fuel/transformers/__init__.py

```diff
@@ -344,12 +431,20 @@ class Batch(Transformer):
         raised if a batch of the requested size cannot be provided.
 
     """
-    def __init__(self, data_stream, iteration_scheme, strictness=0):
+    def __init__(self, data_stream, iteration_scheme, strictness=0, **kwargs):
+        if not data_stream.produces_examples:
+            raise ValueError('the wrapped data stream must produce examples, '
+                             'not batches of examples.')
+        if data_stream.axis_labels:
+            kwargs.setdefault(
+                'axis_labels',
+                dict([(source, ('batch',) + labels if labels else None) for
+                      source, labels in iteritems(data_stream.axis_labels)]))
```

Comment:
In fact, forget what I said about raising an error: there's no need, since you'll get a complaint about duplicate values for keyword anyway. But you should have `produces_examples=False` here.